src/libraries/System.Net.WebSockets.Client/tests/WebSocketHelper.cs

```diff
@@ -13,6 +13,10 @@ public static class WebSocketHelper
     {
         private static readonly Lazy<bool> s_WebSocketSupported = new Lazy<bool>(InitWebSocketSupported);
         public static bool WebSocketsSupported { get { return s_WebSocketSupported.Value; } }
+        // On browser the JavaScript host object 'WebSocket' is only available when DOM is supported
+        // If DOM is not available then a JavaScript error is thrown
+        // --- System.Runtime.InteropServices.JavaScript.JSException : JavaScript host object 'WebSocket' not found.
+        public static bool WebSocketsSupportedInBrowser { get { return !PlatformDetection.IsBrowser || (PlatformDetection.IsBrowser && PlatformDetection.IsBrowserDomSupported); } }
```

Comment:
I don't understand this new check.  WebSocketsSupported supported checks for PNSE which is what we should throw if the platform doesn't support it (and the other tests suggest that we do). If the goal is to only run this test on browser-wasm use PlatformSpecific

Reply:
To expand on my previous comment, if we aren't properly throwing PNSE when the JS runtime doesn't support `WebSocket` we should fix that.  If you want to skip the test on platforms other than Browser use [PlatformSpecific(TestPlatforms.Browser)]. The combination of the two should be enough?